app/controllers/application_controller.rb

```diff
@@ -4,11 +4,17 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :authenticate_user!
   before_action :set_paper_trail_whodunnit
+  before_action :prepare_exception_notifier
 
   helper_method :format_datetime
   
   private
 
+  def prepare_exception_notifier
+    request.env["exception_notifier.exception_data"] = {
+        :current_user => current_user
+    }
```

Comment:
Two concerns here:

1. Does this hash exist before? Do we want to merge rather than overwrite?
2. Hash rockets in flight

Reply:
If I don't initialize the hash then it returns nil in the view. So I don't think it exists before I assign it.

Reply:
That may not necessarily be the case in production, however. Still best to merge rather than overwrite, I think.

Reply:
I mean the key `exception_notifier.exception_data` doesn't exist as far as I know, so I get a `NoMethodError (undefined method `merge' for nil:NilClass)`.

Reply:
We can write code that handles it existing, and not overwriting if it does, and if it doesn't, initializing it with our value.